### ISSUE TITLE
Packets not being sent to the server from client.

### ISSUE THREAD

**Issue by simonglob** [osumitsuha/Ragnarok#1]
I've talked to a few people about this problem, but none of them are able to help as they haven't experienced it themselves. The problem is that some packets aren't being sent to the server from the client. 

Here's an example:
![image](https://user-images.githubusercontent.com/45692977/124330821-79309c80-db8e-11eb-876b-90fefcfc1bc3.png)
*there's also other examples where the packet doesn't queue the first time, but the second time.*

The problem here is that after JOIN_LOBBY packet has been queued, JOIN_CHANNEL should be queued, yet it's not. And it's not Cloudflare caching bancho as I've disabled the proxy. So if you wanna help, you can check the [packets/reader.py](https://github.com/osumitsuha/Ragnarok/blob/dev/packets/reader.py) and [events/bancho.py](https://github.com/osumitsuha/Ragnarok/blob/dev/events/bancho.py#L40) to see if anything might be the cause to this annoying problem.

**Comment by simonglob** [comment]
![image](https://user-images.githubusercontent.com/45692977/124331865-6dde7080-db90-11eb-9452-3f9d9817000f.png)
![screenshot766](https://user-images.githubusercontent.com/45692977/124331889-7c2c8c80-db90-11eb-886f-dca1bf78bcab.jpg)
Here's another example of a packet not being sent. Missing MATCH_COMPLETE packet.

**Comment by cmyui** [comment]
Seems like your reader is only expecting one packet per request, whereas the bancho protocol supports many packets in a single connection.

Basically the packets will be packed side by side, consider `\x00\x00\x00\x0e\x00\x00\x00\x00\x0b\x00\x0b\x00\xc0\x00\x00\x00\x00\x8b\x7f"\x00\x03\x00\x00\x00\x00\x00\x00?\x00\x00\x06\x00\x00\x00\x0b\x04#osu?\x00\x00\x0b\x00\x00\x00\x0b\t#announce`.

This is just a sample request from my client containing 4 packets;
`\x00\x00\x00\x0e\x00\x00\x00\x00\x0b\x00\x0b\x00\xc0\x00\x00\x00\x00\x8b\x7f"\x00` (ChangeAction)
`\x03\x00\x00\x00\x00\x00\x00` (StatsUpdateRequest)
`?\x00\x00\x06\x00\x00\x00\x0b\x04#osu` (ChannelJoin)
`?\x00\x00\x0b\x00\x00\x00\x0b\x04#announce` (ChannelJoin)

It seems like your reader is currently processing the first packet only (reading a 7 byte packet header & packet data), but then there may be more packets (more series of 7 byte packet headers & packet data) after; you should keep reading until there's no data left.

I set up my packet reader as an iterator, so I can do something like 
```py
for packet in BanchoPacketReader(conn.body):
    # once you're ready to handle the packet,
    # simply call it's .handle() method.
    await packet.handle()
```
If you're interested in how that works, [check it out](https://github.com/cmyui/gulag/blob/master/packets.py#L237-L261). If you find it difficult to reason about the packets, I'd recommend printing them out and splitting them up manually like I did above to get a better idea of how they're written, or use a debugger (like vscode's) to manually step through each part of the process.

**Comment by simonglob** [comment]
Thank you so much for your help, cmyui. Fixed in [466579114d5ffc8fc1b47c9a395ce1d2e38ad026](https://github.com/osumitsuha/Ragnarok/commit/466579114d5ffc8fc1b47c9a395ce1d2e38ad026)
